credentials/tls/certprovider/store.go

```diff
@@ -53,6 +54,23 @@ type wrappedProvider struct {
 	store    *store
 }
 
+// closedProvider always returns errProviderClosed error.
+type closedProvider struct{}
+
+func (c closedProvider) KeyMaterial(ctx context.Context) (*KeyMaterial, error) {
+	return nil, errProviderClosed
+}
+
+func (c closedProvider) Close() {
+}
+
+// singleCloseWrappedProvider wraps a provider instance with a reference count to avoid double
+// close still in use provider.
```

Comment:
Nit:

`... with a reference count to properly handle multiple calls to Close.`